apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsCollectionDropdownControl.tsx

```diff
@@ -0,0 +1,70 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import { FormControl, Skeleton } from "@chakra-ui/react"
+import { and, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { FormLabel, SingleSelect } from "@opengovsg/design-system-react"
+import { COLLECTION_DROPDOWN_FORMAT } from "@opengovsg/isomer-components"
+
+import Suspense from "~/components/Suspense"
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+import { siteSchema } from "~/features/editing-experience/schema"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { trpc } from "~/utils/trpc"
+
+export const jsonFormsCollectionDropdownControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.RefControl,
+  and(schemaMatches((schema) => schema.format === COLLECTION_DROPDOWN_FORMAT)),
+)
+
+interface JsonFormsCollectionDropdownControlProps extends ControlProps {
+  data: string
+}
+
+function SuspendableJsonFormsCollectionDropdownControl({
+  data,
+  handleChange,
+  path,
+  label,
+}: JsonFormsCollectionDropdownControlProps) {
+  const { siteId } = useQueryParse(siteSchema)
+
+  const [collections] = trpc.collection.getCollections.useSuspenseQuery({
+    siteId: Number(siteId),
+  })
+
+  return (
+    <SingleSelect
+      value={data}
+      name={label}
+      items={collections.map((collection) => {
+        return {
+          label: collection.title,
+          value: `[resource:${siteId}:${collection.id}]`,
```

Comment:
suggestion: I think there is a util function somewhere that constructs this reference link, would be good to use that instead